### PR TITLE
Add pattern arg to ChosenInlineResultHandler

### LINKED ISSUE #2516: [FEATURE] Add a pattern for result_id of ChosenInlineResultHandler
In this way you can separate the results of your inline queries and redirect them to specific function as it happens for callback queries.

### PR DESCRIPTION
**Pull request by Poolitzer**
### Checklist for PRs

- [x] Added `.. versionadded:: version`, `.. versionchanged:: version` or `.. deprecated:: version` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [x] Created new or adapted existing unit tests
- [ ] Added myself alphabetically to `AUTHORS.rst` (optional)
    
* If relevant:

    - [ ] Added new constants at `telegram.constants` and shortcuts to them as class variables
    - [ ] Added new handlers for new update types
    - [ ] Added new filters for new message (sub)types
    - [ ] Added or updated documentation for the changed class(es) and/or method(s)
    - [ ] Updated the Bot API version number in all places: `README.rst` and `README_RAW.rst` (including the badge), as well as `telegram.constants.BOT_API_VERSION`

This PR closes #2516

### CODE REVIEW

**Comment by Poolitzer** [comment]
I have no idea why codecov/patch fails.

File: telegram/ext/choseninlineresulthandler.py [hunk 1]
```diff
@@ -95,4 +132,22 @@ def check_update(self, update: object) -> Optional[Union[bool, object]]:
...
+        return None
+
+    def collect_additional_context(
+        self,
+        context: 'CallbackContext',
+        update: Update,
+        dispatcher: 'Dispatcher',
+        check_result: Union[bool, Match],
+    ) -> None:
+        if self.pattern:
```
Comment (Bibo-Joshi) [inline telegram/ext/choseninlineresulthandler.py:151]: Let's start adding docstring here :) see #2454, https://github.com/python-telegram-bot/python-telegram-bot/pull/2454/files#diff-5a42014c14adfedb2469c92e0f0df6ebc417f4196c3e4b9e07b1f9190b7b37c0R213

File: telegram/ext/choseninlineresulthandler.py [hunk 2]
```diff
@@ -70,6 +74,11 @@ class ChosenInlineResultHandler(Handler[Update]):
             DEPRECATED: Please switch to context based callbacks.
         run_async (:obj:`bool`): Determines whether the callback will run asynchronously.
             Defaults to :obj:`False`.
+        pattern (:obj:`str` | `Pattern`, optional): Regex pattern. If not :obj:`None`, ``re.match``
+            is used on :attr:`telegram.ChosenInlineResult.result_id` to determine if an update
+            should be handled by this handler.
```
Comment (Bibo-Joshi) [inline telegram/ext/choseninlineresulthandler.py:79]: Maybe add another sentence saying that the match will be available through `context`? Apparently we don't have that for e.g. `CallbackQueryHandler`, but that shouldn't stop us :)

File: telegram/ext/choseninlineresulthandler.py [hunk 3]
```diff
@@ -82,9 +91,37 @@ class ChosenInlineResultHandler(Handler[Update]):
         pass_chat_data (:obj:`bool`): Determines whether ``chat_data`` will be passed to
             the callback function.
         run_async (:obj:`bool`): Determines whether the callback will run asynchronously.
+        pattern (:obj:`str` | `Pattern`): Optional. Regex pattern to test
```
Comment (Bibo-Joshi) [inline telegram/ext/choseninlineresulthandler.py:94]:
```suggestion
        pattern (`Pattern`): Optional. Regex pattern to test
```

File: tests/test_choseninlineresulthandler.py [hunk 4]
```diff
@@ -179,3 +179,33 @@ def test_context(self, cdp, chosen_inline_result):
...
+    def test_with_pattern(self, chosen_inline_result):
+        handler = ChosenInlineResultHandler(self.callback_basic, pattern='.*ult.*')
+
+        assert handler.check_update(chosen_inline_result)
+
+        chosen_inline_result.chosen_inline_result.result_id = 'nothing here'
+        assert not handler.check_update(chosen_inline_result)
+        chosen_inline_result.chosen_inline_result.result_id = 'result_id'
+
+    def callback_context_pattern(self, update, context):
```
Comment (Bibo-Joshi) [inline tests/test_choseninlineresulthandler.py:192]: can you move this up to the top before the first `def test_*`?

**Review by Bibo-Joshi — CHANGES_REQUESTED** [review]
Nice PR! Just a bit of nitpicking :)

**Review by harshil21 — COMMENTED** [review]
nice change.

File: telegram/ext/choseninlineresulthandler.py (hunk 3 above)
Comment (Poolitzer) [reply telegram/ext/choseninlineresulthandler.py:94]: its double in the cbqh as well though, just to defend my choice here

Comment (Bibo-Joshi) [reply telegram/ext/choseninlineresulthandler.py:94]: the `__init__` you wrote tells me that you shouldn'd defend your choice 😉  https://github.com/python-telegram-bot/python-telegram-bot/pull/2517/files#diff-b9b2b3979b2f94c6a992fbfd59df9e5fae154c29832beb89664688452ef38fe8R120-R121

File: telegram/ext/choseninlineresulthandler.py (hunk 2 above)
Comment (Poolitzer) [reply telegram/ext/choseninlineresulthandler.py:79]: I will add it to the doc-fixes PR for the CallbackQueryHandler

Comment (Bibo-Joshi) [reply telegram/ext/choseninlineresulthandler.py:79]: good idea :)

File: telegram/ext/choseninlineresulthandler.py [hunk 5]
```diff
@@ -70,6 +74,12 @@ class ChosenInlineResultHandler(Handler[Update]):
             DEPRECATED: Please switch to context based callbacks.
         run_async (:obj:`bool`): Determines whether the callback will run asynchronously.
             Defaults to :obj:`False`.
+        pattern (:obj:`str` | `Pattern`, optional): Regex pattern. If not :obj:`None`, ``re.match``
+            is used on :attr:`telegram.ChosenInlineResult.result_id` to determine if an update
+            should be handled by this handler. This is accessible in the callback as
+            `context.matches`.
```
Comment (Bibo-Joshi) [inline telegram/ext/choseninlineresulthandler.py:80]:
```suggestion
            ``context.matches``.
```
or even

```suggestion
            :attr:`telegram.ext.CallbackContext.matches`.
```
?

File: telegram/ext/choseninlineresulthandler.py (hunk 2 above)
Comment (Poolitzer) [reply telegram/ext/choseninlineresulthandler.py]: done: e3745fa90971fd250984fa1282192aafda3eb5e0

File: telegram/ext/choseninlineresulthandler.py [hunk 6]
```diff
@@ -95,4 +133,23 @@ def check_update(self, update: object) -> Optional[Union[bool, object]]:
...
+        return None
+
+    def collect_additional_context(
+        self,
+        context: 'CallbackContext',
+        update: Update,
+        dispatcher: 'Dispatcher',
+        check_result: Union[bool, Match],
+    ) -> None:
+        """This function adds the matched regex pattern result to context.matches"""
```
Comment (Bibo-Joshi) [inline telegram/ext/choseninlineresulthandler.py:152]: same as above

File: telegram/ext/choseninlineresulthandler.py (hunk 5 above)
Comment (Poolitzer) [reply telegram/ext/choseninlineresulthandler.py:80]: I thought about the actual link, but for some reason I thought I want to use our naming convention. Having the link generated there is way better though.

**Review by Bibo-Joshi — APPROVED** [review]

**Pull request merged**